### PR TITLE
Module debug draw

### PR DESCRIPTION
**Pull request by mariofv**
What's new:
- New module, `ModuleDebugDraw`. It's responsible of drawing basic shapes (lines, dots, ...).
- Grid, bouding boxes, quadtree and camera frustum are rendered using `ModuleDebugDraw`.
- Delete unnused Renderers, `GeometryRenderer` and `GridRenderer`.

### CODE REVIEW

File: Engine/UI/DebugDraw.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#ifndef _DEBUGDRAW_H_
```
Comment (yalania) [inline Engine/UI/DebugDraw.h:1]: Is this file only for defining types?

File: Engine/Module/ModuleDebugDraw.cpp [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+        glDeleteProgram(textProgram);
+
+        glDeleteVertexArrays(1, &linePointVAO);
+        glDeleteBuffers(1, &linePointVBO);
+
+        glDeleteVertexArrays(1, &textVAO);
+        glDeleteBuffers(1, &textVBO);
+    }
+
+    void setupShaderPrograms()
```
Comment (yalania) [inline Engine/Module/ModuleDebugDraw.cpp]: Could This code about shaders be replace by our ModuleProgram?

File: Engine/Module/ModuleDebugDraw.cpp [hunk 3]
```diff
@@ -0,0 +1,619 @@
...
+    static const char* textFragShaderSrc;
+
+
+}; // class IDebugDrawOpenGLImplementation
+
+// ========================================================
+// Minimal shaders we need for the debug primitives:
+// ========================================================
+
+const char* IDebugDrawOpenGLImplementation::linePointVertShaderSrc = "\n"
```
Comment (yalania) [inline Engine/Module/ModuleDebugDraw.cpp]: Maybe we could move this shaders into a file

File: Engine/Module/ModuleDebugDraw.cpp [hunk 4]
```diff
@@ -0,0 +1,619 @@
+#include "ModuleDebugDraw.h"
+#include "Component/ComponentCamera.h"
+
+#define DEBUG_DRAW_IMPLEMENTATION
+#include "UI/DebugDraw.h"     // Debug Draw API. Notice that we need the DEBUG_DRAW_IMPLEMENTATION macro here!
+
+#include "GL/glew.h"
+#include <assert.h>
+
+class IDebugDrawOpenGLImplementation final : public dd::RenderInterface
```
Comment (yalania) [inline Engine/Module/ModuleDebugDraw.cpp:12]: Maybe we can do the class declaration in the UI/DebugDraw.h  or create another header and define it in a different .cpp

**Review by yalania — COMMENTED** [review]
I suppose you have taken the teacher code as reference but maybe some of the shaders code could be clean up for our own code. But if you find it to troublesome you can leave it like this and maybe we can clean it a little in the future

File: Engine/UI/DebugDraw.h (hunk 1 above)
Comment (mariofv) [reply Engine/UI/DebugDraw.h:1]: It's used as a wrapper of `debug_draw.hpp`. It converts `MathGeoLib` types to required ones and also serves as an entry point to the library.

File: Engine/Module/ModuleDebugDraw.cpp (hunk 2 above)
Comment (mariofv) [reply Engine/Module/ModuleDebugDraw.cpp]: Yes, I will work on that

File: Engine/Module/ModuleDebugDraw.cpp (hunk 3 above)
Comment (mariofv) [reply Engine/Module/ModuleDebugDraw.cpp]: Cool!

File: Engine/Module/ModuleDebugDraw.cpp (hunk 4 above)
Comment (mariofv) [reply Engine/Module/ModuleDebugDraw.cpp:12]: I tried the second approach but it generated conflicts with imports. I might need your help on this.

**Review by yalania — APPROVED** [review]

**Pull request merged**